// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.expenses";
option java_outer_classname = "ExpensesProto";
option objc_class_prefix = "HLW";

import "google/protobuf/timestamp.proto";

package expsenses;

service Expenses {
  // GET all expenses
  rpc GetAllExpenses (ExpensesRequest) returns (ExpensesReply) {}

  // GET one expense
  rpc GetOneExpense (SingleExpenseRequest) returns (SingleExpenseReply) {}
}

message ExpensesRequest {
  string userID = 1;
}

message ExpensesReply {
  repeated SingleExpenseReply expense = 1;
}

message SingleExpenseRequest {
  string userID = 1;
  string expenseID = 2;
}

message SingleExpenseReply {
  string winkelID = 1;
  double price = 2;
  string summary = 3;
  google.protobuf.Timestamp time = 4;
}